client/src/components/MileageModal.vue

```diff
@@ -0,0 +1,175 @@
+            </v-col>
+          </v-row>
+          <v-row>
+            <v-card-text>
+              <v-row>
+                <v-col>
+                  <v-text-field
+                    type="date"
+                    label="Date"
+                    v-model="date"
+                    bg-color="white"
+                    :rules="[required]"
+                  />
+                </v-col>
+              </v-row>
+              <v-row>
+                <v-col>
+                  <p class="px-2 text-subtitle-2">Distance</p>
+                </v-col>
+              </v-row>
+              <v-row>
+                <v-col>
+                  <div v-if="tempIconType === 'running'" class="ma-0 pa-0" id="runner">
+                    <v-slider
+                      v-model="distance"
+                      color="green"
+                      :thumb-size="40"
+                      elevation="0"
+                      :step="1"
+                      min="1"
+                      max="100"
+                    />
+                  </div>
+                  <div v-if="tempIconType === 'walking'" id="walker" class="ma-0 pa-0">
+                    <v-slider
+                      v-model="distance"
+                      color="green"
+                      :thumb-size="40"
+                      elevation="0"
+                      :step="1"
+                      min="1"
+                      max="100"
+                    />
+                  </div>
+                  <div v-if="tempIconType === 'wheeling'" id="wheeler" class="ma-0 pa-0">
+                    <v-slider
+                      v-model="distance"
+                      color="green"
+                      :thumb-size="40"
+                      elevation="0"
+                      :step="1"
+                      min="1"
+                      max="100"
+                    />
+                  </div>
+                </v-col>
+              </v-row>
+              <v-row>
+                <v-col align="center">
+                  <v-chip class="px-6 text-h5 rounded" color="green">{{ distance }}</v-chip>
+                  <p class="text-subtitle-2">KILOMETERS</p>
+                </v-col>
+              </v-row>
+            </v-card-text>
+          </v-row>
+        </v-container>
+        <v-card-actions class="d-flex justify-center">
+          <v-btn @click="$emit('update:modelValue', !modelValue)">CANCEL</v-btn>
+          <v-btn @click="handleSubmit" :disabled="!form" color="primaryRed" variant="elevated"
+            >ADD</v-btn
+          >
+        </v-card-actions>
+      </v-form>
+      <v-spacer />
+    </v-card>
+  </v-dialog>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import { ref, watchEffect } from 'vue'
+
+const isFullscreen = ref(false)
+const props = defineProps(['modelValue'])
+const emit = defineEmits(['update:modelValue', 'handleSubmit'])
+const distance = ref(1)
+const date = ref('')
```

Comment:
make this a mileage object with an inital state of empty

const initialState = { distance: "", date: ""}
const mileage = ref(initalState)